pkg/kube/client/ssa/gvk_parser_cache.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ssa
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/crossplane-contrib/provider-kubernetes/apis/v1alpha1"
+)
+
+// GVKParserCacheManager maintains GVK parser cache stores for each provider config.
+type GVKParserCacheManager struct {
+	// cacheStore holds the *GvkParserCache per provider configuration.
+	// The cacheStore key is the UID of the provider config object.
+	cacheStore map[types.UID]*GvkParserCache
+	// mu is used to make sure the cacheStore map is concurrency-safe.
+	mu *sync.RWMutex
+	// logger is the logger for cacheStore operations.
+	logger logging.Logger
+}
+
+// GVKParserCacheManagerOption lets you configure a *GVKParserCacheManager.
+type GVKParserCacheManagerOption func(cache *GVKParserCacheManager)
+
+// WithCacheStore lets you bootstrap GVKParserCacheManager with your own cacheStore.
+func WithCacheStore(cache map[types.UID]*GvkParserCache) GVKParserCacheManagerOption {
```

Comment:
if this is unused, no need to have it here; I guess we don't expect to provide a stable go vendoring interface and re-usability here

Reply:
done, removed after refactor 👍